msteams-platform/concepts/deploy-and-publish/appsource/prepare/frequently-failed-cases.md

```diff
@@ -1,246 +1,333 @@
->
-> * within your app's description section.
-> * in your app's welcome message.
-> * in your app's help message.
-> * in the window where you ask a user to sign-in to your services.
+The Microsoft 365 App Compliance Program is intended to help organizations assess and manage risk by evaluating security and compliance information about your app.
 
-* Apps without an easy sign-up flow must also include a help tab or link to a web page, where a new user can see detailed guidance on configuring your Teams app. Provide detailed information to ensure a new user is not blocked when trying your app for the first time.
-* Sign-in and sign-out functionality must work on mobile clients. Ensure to use the [Microsoft Teams SDK](https://www.npmjs.com/package/@microsoft/teams-js) version 1.4.1 or later.
+Teams app developers must complete the Publisher Attestation portion of the program. For more information, see the [Microsoft 365 App Compliance Program documentation](https://docs.microsoft.com/microsoft-365-app-certification/docs/attestation).
 
-For additional information on authentication see:
+### 2.2 Financial information
 
-* [Authentication documentation](../../../authentication/authentication.md)
-* [Bot authentication sample in Node](https://github.com/OfficeDev/microsoft-teams-sample-auth-node)
-* [Tab authentication sample in Node](https://github.com/OfficeDev/microsoft-teams-sample-complete-node)
-* [Tab/bot authentication in C#/.NET](https://github.com/OfficeDev/microsoft-teams-sample-complete-csharp)
+Your app must not ask users to make payments within the Teams interface. Financial instrument details must not be transmitted through the user to the bot interface. Links to secure payment services may only be transmitted to users if disclosure has been made in the app’s terms of use, privacy policy, or any profile page or website before the user agrees to use the bot.
 
-### &#9989; Response times must be reasonable
+### 2.3 Bot Framework
 
-* **Tabs**. If a response to an action takes more than three seconds, you must provide a loading message or warning.
-* **Bots**. A response to a user command must occur within two seconds. If longer processing is required, your app must display a typing indicator.
-* **Compose extensions**. A response to a user command must occur within five seconds.
+For apps with bots or messaging extensions, you must follow all requirements (including privacy notice requirements) as defined in the Microsoft Bot Framework Developer Code of Conduct and Online Services Agreement.
 
-> [!TIP]
-> Make sure your app displays a loading indicator or some form of warning when your app is taking longer than expected to respond.
+Bots must always ask permission to upload a file and display a confirmation message after the file is uploaded.
+
+### 2.4 External domains
+
+Tunneling services and domains outside of an organization’s control, including wildcards, must not be included in the valid domains of the app manifest.
```

Comment:
Update - Domains outside of your organization's control (including wildcards) and tunneling services cannot be included in the valid domains of your manifest, except in the following conditions:

If you are using OAuthCard, Token.botframework.com must be in the valid domains list.
Teams apps that require their own SharePoint URLs to function may include {teamsitedomain} in their valid domain list.